datastream-kafka/src/test/java/com/linkedin/datastream/kafka/TestKafkaTransportProvider.java

```diff
@@ -205,6 +205,86 @@ public void testEventWithoutKeyValueAndPartition() throws Exception {
     testEventSend(1, 2, -1, false, false, "test");
   }
 
+  @Test
+  public void testSendMultipleEventsInSingleDatastreamProducerRecord() throws Exception {
+    String metricsPrefix = "test";
+    final int numberOfEvents = 10;
+    String topicName = getUniqueTopicName();
+    final int[] eventsReceived = {0};
+
+    _transportProviderProperties.put(KafkaTransportProviderAdmin.CONFIG_METRICS_NAMES_PREFIX, metricsPrefix);
+    KafkaTransportProviderAdmin provider = new KafkaTransportProviderAdmin("test", _transportProviderProperties);
+
+    String destinationUri = provider.getDestination(null, topicName);
+
+    Datastream ds = DatastreamTestUtils.createDatastream("test", "ds1", "source", destinationUri, 1);
+
+    DatastreamTask task = new DatastreamTaskImpl(Collections.singletonList(ds));
+    TransportProvider transportProvider = provider.assignTransportProvider(task);
+    provider.createTopic(destinationUri, 1, new Properties());
+
+    KafkaTestUtils.waitForTopicCreation(_zkUtils, topicName, _kafkaCluster.getBrokers());
+
+    LOG.info(String.format("Topic %s created with %d partitions and topic properties %s", topicName, 1,
+        new Properties()));
+    DatastreamProducerRecord event =
+        createEventsForSingleDatastreamProducerRecord(topicName, 0, numberOfEvents, true, true);
+    Assert.assertEquals(event.getEvents().size(), numberOfEvents);
+
+    LOG.info(String.format("Trying to send %d events to topic %s", event.getEvents().size(), topicName));
+
+    final Integer[] callbackCalled = {0};
+    List<Integer> indexList = new ArrayList<>();
+    transportProvider.send(destinationUri, event, ((metadata, exception) -> {
+      callbackCalled[0]++;
+      indexList.add(metadata.getBrooklinEventIndex());
+    }));
+
+    // wait until all messages were acked, to ensure all events were successfully sent to the topic
+    Assert.assertTrue(PollUtils.poll(() -> callbackCalled[0] == event.getEvents().size(), 1000, 10000),
```

Comment:
You can rely on `indexList.size()` instead of `callbackCalled[0]`

Reply:
done